packages/cdktf-cli/lib/cdktf-project.ts

```diff
@@ -279,17 +279,22 @@ export type MultipleStackOptions = {
   stackNames?: string[];
 };
 
-export type ExecutionOptions = MultipleStackOptions & {
+export type AutoApproveOptions = {
   autoApprove?: boolean;
-  ignoreMissingStackDependencies?: boolean;
-  parallelism?: number;
-  refreshOnly?: boolean;
 };
 
 export type DiffOptions = SingleStackOptions & {
   refreshOnly?: boolean;
 };
 
+export type MutationOptions = MultipleStackOptions &
```

Comment:
I'm not in love with the name, and after all the modifications, I wonder if the name change is even required. I chose mutation because it was going to be consumed by both `deploy` and `destroy`, which are both operations that mutate state, as opposed to Synth.

Reply:
If we go with that, we should rename the `execute` method as well, as I think that one was the original reason for it to be named `ExecutionOptions`. I personally would also be fine with a verbose `DeployDestroyOptions` 😁 But Mutation seems like a valid umbrella term for both of those.

What do you think, @DanielMSchmidt? You came up with this name originally, is that wording maybe already used in the Terraform world for apply/destroy?

Reply:
I'm fine with any option here, naming things is hard and mutation fits quite ok. My thought was execution but for no deeper reason than I needed a term that fits these two